### PR TITLE
Add ViewComponentResult helper methods to Page \ PageModel

### LINKED ISSUE #7051: Add ViewComponentResult helper methods to Page \ PageModel
In the underlying `PageModel` i want to return a ViewComponent from a OnGet request. I am trying return ViewComponent("ComponentName"); but it throws an error `non-invocable member ViewComponent`

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #7051

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs [hunk 1]
```diff
@@ -149,7 +149,6 @@ public override void EndContext()
         /// Creates a <see cref="BadRequestResult"/> that produces a <see cref="StatusCodes.Status400BadRequest"/> response.
         /// </summary>
         /// <returns>The created <see cref="BadRequestResult"/> for the response.</returns>
-        [NonAction]
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs]: @Eilon is removing an attribute a breaking change?

Comment (Eilon) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs]: In this case it has no real effect, right? I think that's fine. Only in the craziest case where someone derived from `PageBase` and used the type **as a POCO controller**, would anything behave differently. Too bad for those people.

**Comment by pranavkm** [comment]
@dotnet-bot test Windows Release x64 Build please

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs [hunk 2]
```diff
@@ -1617,6 +1614,78 @@ public virtual ObjectResult StatusCode(int statusCode, object value)
...
+        /// An <see cref="object"/> with properties representing arguments to be passed to the invoked view component
+        /// method. Alternatively, an <see cref="System.Collections.Generic.IDictionary{String, Object}"/> instance
+        /// containing the invocation arguments.
+        /// </param>
+        /// <returns>The created <see cref="ViewComponentResult"/> object for the response.</returns>
+        public virtual ViewComponentResult ViewComponent(Type componentType, object arguments)
+        {
+            return new ViewComponentResult
+            {
+                ViewComponentType = componentType,
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: null checks?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: This is verbatim from Controller - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs#L274-L283

**Pull request merged**